osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBall.cs

```diff
@@ -76,26 +78,59 @@ public SliderBall(Slider slider)
             };
         }
 
-        private InputState lastState;
+        private Vector2 lastPosition;
+        private bool actionLeftButtonPressed;
+        private bool actionRightButtonPressed;
 
         protected override bool OnMouseDown(InputState state, MouseDownEventArgs args)
         {
-            lastState = state;
+            if (state != null)
+                lastPosition = state.Mouse.NativeState.Position;
             return base.OnMouseDown(state, args);
         }
 
         protected override bool OnMouseUp(InputState state, MouseUpEventArgs args)
         {
-            lastState = state;
+            if (state != null)
+                lastPosition = state.Mouse.NativeState.Position;
             return base.OnMouseUp(state, args);
         }
 
         protected override bool OnMouseMove(InputState state)
         {
-            lastState = state;
+            if (state != null)
+                lastPosition = state.Mouse.NativeState.Position;

```

Comment:
Since you're doing this in MouseMove, the ones in MouseDown and MouseUp aren't needed.